sql/partitioning/partition_base.cc

```diff
@@ -0,0 +1,4683 @@
+
+  @return Operation status
+    @retval >0  Error
+    @retval 0   Success
+
+  @note  Common routine to handle delete_table and rename_table.
+  The routine uses the partition handler file to get the
+  names of the partition instances. Both these routines
+  are called after creating the handler without table
+  object and thus the file is needed to discover the
+  names of the partitions and the underlying storage engines.
+*/
+
+int Partition_base::del_ren_table(const char *from, const char *to,
+                                  const dd::Table *table_def_from,
+                                  dd::Table *table_def_to) {
+  int error = 0;
+  int save_error = 0;
+  const char *from_path;
+  const char *to_path = nullptr;
+  char from_lc_buff[FN_REFLEN];
+  char to_lc_buff[FN_REFLEN];
+  std::unique_ptr<handler, Destroy_only<handler>> file(
+      get_file_handler(nullptr, ha_thd()->mem_root));
+  /*
+    Since Partition_base has HA_FILE_BASED, it must alter underlying table names
+    if they do not have HA_FILE_BASED and lower_case_table_names == 2.
+    See Bug#37402, for Mac OS X.
+    The appended #P#<partname>[#SP#<subpartname>] will remain in current case.
+    Using the first partitions handler, since mixing handlers is not allowed.
+  */
+  from_path = get_canonical_filename(file.get(), from, from_lc_buff);
+  if (to != nullptr)
+    to_path = get_canonical_filename(file.get(), to, to_lc_buff);
+
+  std::vector<std::string> from_names;
+  std::vector<std::string> to_names;
+
+  for (const dd::Partition *dd_part : table_def_from->leaf_partitions()) {
+    char name_buff[FN_REFLEN];
+    part_name(name_buff, from_path,
+              dd_part->parent() ? dd_part->parent()->name().c_str() : nullptr,
+              dd_part->name().c_str());
+    from_names.push_back(name_buff);
+  }
+
+  if (to)
+    for (const dd::Partition *dd_part :
+         const_cast<const dd::Table *>(table_def_to)->leaf_partitions()) {
+      char name_buff[FN_REFLEN];
+      part_name(name_buff, to_path,
+                dd_part->parent() ? dd_part->parent()->name().c_str() : nullptr,
+                dd_part->name().c_str());
+      to_names.push_back(name_buff);
+    }
+
+  auto to_name = to_names.begin();
+
+  for (const std::string &from_name : from_names) {
+    if (!to) {
+      error = file->ha_delete_table(from_name.c_str(), table_def_from);
+      if (error) save_error = error;
+    } else {
+      // TODO: fix nullptr here. Currently tokudb does not process dd::Table
+      // object, look at ha_innopart::rename_table() to understand
+      // how to modify table_def_to
+      error = file->ha_rename_table(from_name.c_str(), (to_name++)->c_str(),
+                                    nullptr, nullptr);
+
+      if (error) goto rename_error;
+    }
+  }
+
+  return save_error;
+
+rename_error:
+  to_name = to_names.begin();
+  for (const std::string &from_name : from_names) {
+    // TODO: fix nullptr here. Currently tokudb does not process dd::Table
```

Comment:
Log a JIRA issue if not done already?

Reply:
There is no need in JIRA issue, the comment is fixed as discussed offline.